bindings/matlab/Visualization/getMeshes.m

```diff
@@ -0,0 +1,108 @@
+function [linkMeshInfo,map]=getMeshes(model,meshFilePrefix)
+% We use the iDyntree information to obtain the files containing the meshes
+% and we link them to their link names
+% Gets the mesh information for each link in the model.
+%     - Iputs:
+%         - `model` : iDyntree model loaded form a URDF.
+%         - `meshFilePrefix` : Path in which we can find the meshes. As an example the path to the mesh in a iCub urdf is `'package://iCub/meshes/simmechanics/sim_sea_2-5_root_link_prt-binary.stl'
+% `. `meshFilePrefix` should replace package to allow to find the rest of the path.
+%     - Outputs:
+%         - `map`        : Cell array having both the names of the meshes and the associated link
+%         - `linkMeshInfo` : Struct array that contain the link name and a struct (`meshInfo`) that contains the name of file or if is a simple geometry, the triangulation ( edges and vertices of the mesh ) and the link to geometry transform.
+%
+% Author : Francisco Andrade (franciscojavier.andradechavez@iit.it)
+%
+% Copyright (C) 2019 Istituto Italiano di Tecnologia (IIT). All rights reserved.
+% This software may be modified and distributed under the terms of the
+% GNU Lesser General Public License v2.1 or any later version.
+
+
+
+% get the linkSolidShapes containing the mesh information
+visual=model.visualSolidShapes;
+linkSolidShapesV=visual.linkSolidShapes;
+% get number of links
+numberOfLinks=linkSolidShapesV.size;
+linkMeshInfo=struct('meshInfo',{},'linkName',{});
+% get pointer to beggining of vector
+iterator=linkSolidShapesV.begin;
+% iterate getting the name of the mesh and the name of the link
+count=1;
+link_with_no_visual=[];
+for links=1:numberOfLinks    
+    linkName=model.getLinkName(links-1);
+    linkMeshInfo(links).linkName=linkName;
+    solidarray=iterator.next;
+    solids_number=size(solidarray,2);    
+    meshInfo=struct('meshFile',{},'mesh_triangles',{},'link_H_geom',{});
+    for solids=1:solids_number
+        if solidarray{solids}.isExternalMesh
+            meshName=split(solidarray{solids}.asExternalMesh.filename,':');
+            meshFile=meshName{2};
+            % Import an STL mesh, returning a PATCH-compatible face-vertex structure
+            mesh_triangles = stlread([meshFilePrefix meshFile]);
```

Comment:
A URDF file can also contain absolute files (that is for example how the current visualizer works), so before appending a prefix you should check if the `filename` does not contain an absolute path.

Reply:
Can you put an example? That way I can see how to distinguish relative from absolute path. I  guess one way is to check if the file alone exists and if not use the prefix.

Reply:
See https://github.com/robotology/idyntree/issues/291 . I guess that if a file starts with `package://`  or `file://` for sure it is  not a relative path.

Reply:
Actually we are  tackling only the package case. Sibcce I do not have the resource finder. I'm directly asking for the path that would replace package. So in that way we are actually handling absolute paths and not relative paths.

Reply:
Yes, but in this line https://github.com/robotology/idyntree/pull/668/files#diff-34f42ee1b4d63ad197c2d3194a22bacfR42 you need to check if the initial part of the URI is `package://`, otherwise you are going to add the prefix also to absolute files (specified by `file://`).

Reply:
Thanks, I was missing the piece of information between package and file, I added now a check for the keyword package and only add the mesh path in that case. Is there other specific words used? The only relevant reference I found with the `file://` notation is [this](https://answers.gazebosim.org//question/12387/referencing-models-from-other-packages-launch-files/) and it mentions also the `model://` option. Do you have some documentation where this keywords are mentioned?